### PR TITLE
Cache hierarchy rework, AppSettings builder

### PR DESCRIPTION
**Pull request by lobeiko-oleg-351002**

### CODE REVIEW

**Comment by lobeiko-oleg-351002** [comment]
I left ICacheService<StudentViewModel> as returning type in Resolver for now, but I wonder what's the proper solution for resolving and adding several types of cache services.

File: BLL/Caching/UserCacheService.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using SchedulerViewModels;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BLL.Caching
+{
+    public class UserCacheService : CacheService<StudentViewModel>, ICacheService<StudentViewModel>
```
Comment (sinelnykov) [inline BLL/Caching/UserCacheService.cs:11]: delete it

File: SchedulerApp/Caching/CacheTypeResolver.cs [hunk 2]
```diff
@@ -13,23 +16,25 @@ namespace SchedulerApp.Caching
     public class CacheTypeResolver
     {
         private readonly IServiceProvider _serviceProvider;
+        private readonly AppSettings _appSettings;
 
-        public CacheTypeResolver(IServiceProvider serviceProvider)
+        public CacheTypeResolver(IServiceProvider serviceProvider, AppSettings appSettings)
```
Comment (sinelnykov) [inline SchedulerApp/Caching/CacheTypeResolver.cs:21]: please create an interface IAppSettings and bind it as Singleton 
so you will call GetAppSettings one time only

File: BLL/Caching/Base/RedisCacheService.cs [hunk 3]
```diff
@@ -8,44 +8,49 @@
...
+            var userData = _cache.Get(key);
             try
             {
-                return DecodeObject(userData);
+                return DecodeObject<T>(userData);
             }
             catch
             {
-                return null;
+                return default(T);
```
Comment (sinelnykov) [inline BLL/Caching/Base/RedisCacheService.cs:33]: just throw;

**Review by sinelnykov — APPROVED** [review]

**Pull request merged**